src/main/webapp/js/ideas/ideaUpdate.controller.js

```diff
@@ -5,20 +5,88 @@
         .module('app.controllers')
         .controller('updateIdea', updateIdea);
 
-    updateIdea.$inject = ['$scope', '$window', /*'restFactory', */'ideasFactory', 'detailsService', 'ideaDetails', 'mapGeoService'];
+    updateIdea.$inject = ['$scope', '$window', '$modal', 'ideasFactory', 'detailsService', 'ideaDetails', 'mapGeoService', 'imgur', 'Upload'];
 
-    function updateIdea($scope, $window, /*restFactory, */ideasFactory, detailsService, ideaDetails, mapGeoService) {
+    function updateIdea($scope, $window, $modal, ideasFactory, detailsService, ideaDetails, mapGeoService, imgur, Upload) {
 
         this.categories =  detailsService.getCategories();
         $scope.bottomButtonName = 'Обновить';
         $scope.idea = ideaDetails;
         $scope.data = null;
 
+        $scope.imageExist = false;
+        $scope.imageFile;
+        $scope.imageUrl = null;
+        $scope.modalInstance = null;
+        $scope.files = null;
+
         //maps data
 		$scope.center = [30.331014, 53.894617];
 		var map = null;
 		var ideaCoords = null;
 
+		$scope.openModalWindow = function(){
+			$scope.modalInstance = $modal.open({
+				animation: true,
+				templateUrl: 'myModalContent.html',
+				controller: 'loadModalWindow',
+				size: 'lg'/*,
+				resolve: {
+				  caption: function () {
+					return $scope.caption;
+				  }
+				}*/
+			  });
+		};//openModalWindow
+
+		$scope.closeModalWindow = function(){
+			//$scope.modalInstance.dismiss('cancel');
+			$scope.modalInstance.close();
+			$scope.modalInstance = null;
+		}//closeModalWindow
+
+		$scope.$watch('files', function(){
+			if ($scope.files && $scope.files.length) {
+				$scope.openModalWindow();
+				$scope.load2Imgur($scope.files);
+			}//if
+		});
+
+		$scope.chooseFile = function(){
+    		$('input[type=file]').click();
+
+			$('input[type=file]').change(function() {
+				//read file
+				var input = document.getElementById('imageLoader');
+
+				if(input && input.files && input.files[0]){
+					$scope.imageFile = input.files[0];
+
+					$scope.openModalWindow();
+					$scope.load2Imgur($scope.imageFile);
+				}//if
+			});
+		};//chooseFile
+
+		$scope.load2Imgur = function(file){
+			//load file to imgur
+			imgur.setAPIKey('Client-ID c62cfae02efe4c0');
+			imgur.upload(file).then(function then(model) {
+					console.log('Your adorable cat be here: ' + model.link);
+
+					if(Object.getPrototypeOf(model) === Object.prototype){
+						$scope.imageUrl = model.link;
+					}else{
+						$scope.imageUrl = model[0].link;
+					}
+
+					$scope.imageExist = true;
+
+					//hide window #modalWindow
+					$scope.closeModalWindow();
+			});
+		};//load2Imgur
```

Comment:
it's use template in html and I not sure can it work when will call from service